engine/apps/user_management/models/user.py

```diff
@@ -169,6 +169,12 @@ class Meta:
         # Including is_active to unique_together and setting is_active to None allows to
         # have multiple deleted users with the same user_id, but user_id is unique among active users
         unique_together = ("user_id", "organization", "is_active")
+        indexes = [
+            models.Index(fields=["organization", "is_active", "username", "role"]),
+            models.Index(fields=["organization", "is_active", "email", "role"]),
+            models.Index(fields=["username", "organization", "is_active", "permissions"]),
+            models.Index(fields=["email", "organization", "is_active", "permissions"]),
```

Comment:
Any reason for ordering fields differently? (ie. `"organization", "is_active", "username"` vs `"username", "organization", "is_active"`)

+1 to add some indexing to the User model, wondering if keeping `"role"` and `"permissions"` out wouldn't be enough (thinking on space)? so we only add:
`"organization", "is_active", "username"`
`"organization", "is_active", "email"`

Reply:
The order of the indexing matters for performance reasons depending on the SQL engine.
Cardinality matters as it reduces the search time for rows, which this is attempting to eliminate as many rows as possible to remove the very costly table scan.

All 4 indexes are required as they're effectively used as a set of joins to reduce the number of rows down to (preferably) 1 row (in the best case).